client/state/notices/reducer.js

```diff
@@ -31,6 +32,16 @@ export const items = createReducer( {}, {
 
 		return omit( state, noticeId );
 	},
+	[ NOTICE_CLEAR ]: ( state ) => {
+		return reduce( state, ( memo, notice, noticeId ) => {
```

Comment:
@youknowriad Would we ever want to have multiple notices within this same group? Otherwise I think we could just leverage the fact that we can [specify a custom `id`](https://github.com/Automattic/wp-calypso/blob/e25ca42dba72fda1b972cf2bb9cb3e5a4a656cc9/client/state/notices/actions.js#L23) when creating a notice and pass that same ID in the [`removeNotice` action creator](https://github.com/Automattic/wp-calypso/blob/e25ca42dba72fda1b972cf2bb9cb3e5a4a656cc9/client/state/notices/actions.js#L14).